src/key.cpp

```diff
@@ -272,22 +272,17 @@ bool CKey::SignCompact(const uint256 &hash, std::vector<unsigned char>& vchSig)
 bool CKey::SignSchnorr(const uint256& hash, Span<unsigned char> sig, const uint256* merkle_root, const uint256& aux) const
 {
     assert(sig.size() == 64);
-    secp256k1_keypair keypair;
-    if (!secp256k1_keypair_create(secp256k1_context_sign, &keypair, UCharCast(begin()))) return false;
-    if (merkle_root) {
-        CPubKey pubkey_bytes = GetPubKey();
-        uint256 tweak = XOnlyPubKey(pubkey_bytes).ComputeTapTweakHash(merkle_root->IsNull() ? nullptr : merkle_root);
-        if (!secp256k1_keypair_xonly_tweak_add(secp256k1_context_static, &keypair, tweak.data())) return false;
-    }
-    bool ret = secp256k1_schnorrsig_sign32(secp256k1_context_sign, sig.data(), hash.data(), &keypair, aux.data());
+    KeyPair kp = ComputeKeyPair(merkle_root);
```

Comment:
nitty-nit:
I think `ConstructKeyPair` might be a better name?

Reply:
I prefer to use `Compute*` whenever an actual computation is happening. In this case, calling `ComputeKeyPair` will compute a public key for the private key and (optionally) tweak the newly created key pair.